app/models/miq_server/role_management.rb

```diff
@@ -242,32 +249,32 @@ def synchronize_active_roles(servers, roles_to_sync)
         delta    = r.max_concurrent - active.length
         if delta < 0
           delta.abs.times do
-            if active.length > 0
-              s, p = active.shift
-              s.deactivate_roles(role_name)
-              inactive << [s, p]
-            end
+            next if active.empty?
+
+            s, p = active.shift
+            s.deactivate_roles(role_name)
+            inactive << [s, p]
           end
           inactive = inactive.sort_by(&:last) # Sort again, since we may have added to array
         elsif delta > 0
           delta.times do
-            if inactive.length > 0
-              s, p = inactive.shift
-              s.activate_roles(role_name)
-              active << [s, p]
-            end
+            next if inactive.empty?
+
+            s, p = inactive.shift
+            s.activate_roles(role_name)
+            active << [s, p]
           end
           active = active.sort_by(&:last).reverse # Sort again, since we may have added to array
         end
 
         active.each do |s, p|
-          if (inactive.length > 0) && (p > inactive.first.last)
-            s2, p2 = inactive.shift
-            _log.info("Migrating Role <#{role_name}> Active on Server <#{s.name}> with Priority <#{p}> to Server <#{s2.name}> with Priority <#{p2}>")
-            s.deactivate_roles(role_name)
-            s2.activate_roles(role_name)
-            active << [s2, p2]
-          end
+          next unless !inactive.empty? && (p > inactive.first.last)
```

Comment:
I'm fine with this change although I generally prefer not using multiple expressions with `unless` but that's just me.

Reply:
:+1: -  unless with multiple expression becomes impossible to read.  Throw in a negative variable name (inactive), with a `!` AND  an `empty?` (quadruple negation), and I'm lost.
```suggestion
          next if inactive.empty? || p > inactive.first.last
```

I _think_ this is the same.  Hopefully there are specs on this.

Reply:
For now, I would take this one out of this PR, since it's a bit too complicated.

Reply:
Restored.